### PR TITLE
[ONNX] replace all _C._flatten to torch.jit._flatten

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83186
* #83201
* #83599
* __->__ #83598

_C._flatten is exactly the same as torch.jit._flatten. Unifying them to reduce confusion.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1201,7 +1201,7 @@ def test_arg_configurations(self):
 
         @torch.jit.compile(nderivs=0)
         def fn(*args):
-            in_vars, _ = torch._C._jit_flatten(args)
+            in_vars, _ = torch.jit._flatten(args)
```
Comment (justinchuby) [inline test/test_jit.py:1204]: Looks like `ONNX exporter` doesn't have merge permission on this file. changing it can take longer

**Review by justinchuby — APPROVED** [review]

File: test/test_jit.py (hunk 1 above)
Comment (AllenTiTaiWang) [reply test/test_jit.py:1204]: I can change this one back.

**Review by BowenBao — APPROVED** [review]

File: test/test_jit.py (hunk 1 above)
Comment (AllenTiTaiWang) [reply test/test_jit.py]: Done

**Pull request closed without merging**